tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/DecisionTreeTrainer.scala

```diff
@@ -170,10 +168,23 @@ private class Node(data: Option[FeatureVectorSource], val featureVectorSubset: S
   }
 }
 
-/** Functions for training decision trees. */
-//param validationPercentage percentage of the training vectors to hold out for validation
+sealed trait InformationGainMetric {
+  val minimumGain: Double
+}
+case class EntropyGainMetric(minimumGain: Double) extends InformationGainMetric
```

Comment:
In that case, you could make these `object` instead of `class`, which saves having to type parens in the `match` expression. ;)